test/inference/tempering.jl

```diff
@@ -0,0 +1,26 @@
+@testset "tempering.jl" begin
+    @turing_testset "sampling" begin
+        Random.seed!(0)
+        N = 1000
+        samplers = (
+            HMC(0.1, 7),
+            MH(),
+            NUTS(0.65)
+        )
+        for s in samplers
+            ts = Tempered(s, 4)
```

Comment:
A Turing-unrelated comment: IMO the uppercase `T` suggests incorrectly that this is a constructor and `ts isa Tempered`. Usually, lowercase `tempered` is used in such situations. E.g., compare `Truncated` and `truncated` in Distributions: the first one always creates an object of type `Truncated` whereas the latter returns a truncated distribution that possibly is optimized for the unconstrained distribution and only falls back to `Truncated` if no better implementation exists. Users are explicitly advised to not use `Truncated`.

Reply:
That is a fair point, will perhaps change the functional form to `tempered` then in next MCMCTempering release. More tests to come by the way, been having technical issues for the past day so have some more stuff locally